src/lib/actions/RuleDescribeAction.ts

```diff
@@ -0,0 +1,94 @@
+import {Action} from "../ScannerCommand";
+import {Inputs, Rule} from "../../types";
+import {AnyJson} from "@salesforce/ts-types";
+import {RuleFilter} from "../RuleFilter";
+import {Controller} from "../../Controller";
+import {BundleName, getMessage} from "../../MessageCatalog";
+import {deepCopy} from "../util/Utils";
+import Dfa from "../../commands/scanner/run/dfa";
+import Run from "../../commands/scanner/run";
+import {Ux} from "@salesforce/sf-plugins-core";
+import {Display} from "../Display";
+import {RuleFilterFactory} from "../RuleFilterFactory";
+
+type DescribeStyledRule = Rule & {
+	runWith: string;
+	enabled: boolean;
+};
+
+/**
+ * The Action behind the "rule describe" command
+ */
+export class RuleDescribeAction implements Action {
+	private readonly display: Display;
+	private readonly ruleFilterFactory: RuleFilterFactory;
+
+	public constructor(display: Display, ruleFilterFactory: RuleFilterFactory) {
+		this.display = display;
+		this.ruleFilterFactory = ruleFilterFactory;
+	}
+
+	public async validateInputs(_inputs: Inputs): Promise<void> { // eslint-disable-line @typescript-eslint/no-unused-vars
+		// Currently there is nothing to validate
+		return Promise.resolve();
+	}
+
+	public async run(inputs: Inputs): Promise<AnyJson> {
+		const jsonEnabled: boolean = inputs.json as boolean;
+
+		const ruleFilters: RuleFilter[] = this.ruleFilterFactory.createRuleFilters(inputs);
+
+		// TODO: Inject RuleManager as a dependency to improve testability by removing coupling to runtime implementation
+		const ruleManager = await Controller.createRuleManager();
+
+		const rules: DescribeStyledRule[] = await this.styleRules(ruleManager.getRulesMatchingOnlyExplicitCriteria(ruleFilters));
+		if (rules.length === 0) {
+			// If we couldn't find any rules that fit the criteria, we'll let the user know. We'll use .warn() instead of .log()
```

Comment:
This comment should be updated.

Reply:
ok good find.

Reply:
Fixed - squashed into part 10 with rebase and then force push.